extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -404,10 +404,16 @@ export class AmpA4A extends AMP.BaseElement {
       // onLayoutMeasure gets called multiple times.
       return;
     }
-    this.layoutMeasureExecuted_ = true;
+    const slotRect = this.getIntersectionElementLayoutBox();
+    if (slotRect.height == 0 || slotRect.width == 0) {
```

Comment:
Then maybe the thing to do is to call [`isDisplayed`](https://github.com/ampproject/amphtml/blob/master/src/service/resource.js#L484) directly, rather than duplicating the code?  Also, I worry that that logic is special-purpose and isn't intended to cover CSS styling `display:none`.